libs/stocks/feature-shell/src/lib/stocks/stocks.component.ts

```diff
@@ -25,19 +29,99 @@ export class StocksComponent implements OnInit {
     { viewValue: 'One month', value: '1m' }
   ];
 
+  get shouldDisplayDateError(): boolean {
+    return (
+      this.stockPickerForm.errors && this.stockPickerForm.errors.fromAfterTo
+    );
+  }
+
   constructor(private fb: FormBuilder, private priceQuery: PriceQueryFacade) {
-    this.stockPickerForm = fb.group({
-      symbol: [null, Validators.required],
-      period: [null, Validators.required]
-    });
+    const today = new Date();
+    const year = today.getFullYear();
+    const month = today.getMonth();
+    const date = today.getDate();
+    // according to API documentation the oldest records will be 15 years old
+    // https://iexcloud.io/docs/api/#charts
+    const maxAgeOfAPIData = 15;
+
+    this.maxDate = new Date(year, month, date);
+    this.minDate = new Date(year - maxAgeOfAPIData, month, date);
+
+    this.stockPickerForm = fb.group(
+      {
+        symbol: [null, Validators.required],
+        to: [today, [Validators.required]],
+        from: [today, [Validators.required]]
+      },
+      { validators: DateRangeValidator }
+    );
+
+    this.stockPickerForm.valueChanges
+      .pipe(
+        debounceTime(500),
+        filter(() => this.stockPickerForm.invalid),
+        map(() => {
+          if (this.shouldDisplayDateError) {
+            this.stockPickerForm.get('from').patchValue(today);
+            this.stockPickerForm.get('to').patchValue(today);
+            this.stockPickerForm.updateValueAndValidity();
+          }
+        })
+      )
+      .subscribe();
   }
 
   ngOnInit() {}
 
   fetchQuote() {
     if (this.stockPickerForm.valid) {
-      const { symbol, period } = this.stockPickerForm.value;
-      this.priceQuery.fetchQuote(symbol, period);
+      const { symbol, to, from } = this.stockPickerForm.value;
+      const necessaryPeriod = this.setPeriod(from);
+
+      this.priceQuery.fetchQuote(symbol, necessaryPeriod, from, to);
     }
   }
+
+  private setPeriod(from: Date): string {
+    let desiredSearchPeriod = 'max';
+    const today = new Date();
+    const year = today.getFullYear();
+    const month = today.getMonth();
+    const date = today.getDate();
+    const fiveYearsAgo = new Date(year - 5, month, date);
```

Comment:
This jumping through hoops is to prevent edge case errors for timestamps with minutes and seconds